turbinia/workers/analysis/ssh_analyzer.py

```diff
@@ -0,0 +1,397 @@
+            f' in {log_filename}')
+
+    ssh_records = []
+
+    sshd_message_type_re = re.compile(
+        r'.*sshd\[\d+\]:\s+([^\s]+)\s+([^\s]+)\s.*')
+
+    # Only processing authentication logs with pattern sshd[9820]
+    for line in re.findall(r'.*sshd\[\d+\].*', data):
+      try:
+        sshd_message_type = sshd_message_type_re.search(line).group(1)
+      except AttributeError:
+        # NOTE: This does not mean actual error. This means
+        log.error(f'Unable to get SSH message type: {line}')
+        continue
+
+      for key, value in self.MESSAGE_GRAMMAR.items():
+        if key.lower() == sshd_message_type.lower():
+          try:
+            log.debug(f'Procesing log line {line}')
+            m = value.parseString(line)
+
+            # handle date/time
+            dt_object = self.parse_message_datetime(m.datetime, log_year)
+            event_date = dt_object.strftime('%Y-%m-%d')
+            event_time = dt_object.strftime('%H:%M:%S')
+            event_timestamp = dt_object.timestamp()
+
+            # event_type and auth_result
+            if key.lower() == 'accepted':
+              event_type = 'authentication'
+              auth_result = 'success'
+            elif key.lower() == 'failed':
+              event_type = 'authentication'
+              auth_result = 'failure'
+            elif key.lower() == 'disconnected':
+              event_type = 'disconnection'
+              auth_result = ''
+            else:
+              event_type = 'unknown'
+              auth_result = ''
+
+            ssh_event_data = SSHEventData(
+                timestamp=event_timestamp, date=event_date, time=event_time,
+                hostname=m.hostname, pid=m.pid, event_key=event_type,
+                event_type=event_type, auth_method=m.auth_method,
+                auth_result=auth_result, username=m.username,
+                source_hostname='', source_ip=m.source_ip,
+                source_port=m.source_port)
+            ssh_event_data.calculate_session_id()
+
+            ssh_records.append(ssh_event_data)
+          except pyparsing.ParseException as e:
+            log.debug(f'Pyparsing parsing exception: {str(e)}')
+            continue
+
+    log.info(
+        f'Total number of SSH records {len(ssh_records)} in {log_filename}')
+    return ssh_records
+
+  def get_priority_value(self, priority_string: str) -> Priority:
+    """Return priority value"""
+    analyzer_priority_string = priority_string.upper()
+
+    if analyzer_priority_string == 'CRITICAL':
+      return Priority.CRITICAL
+    elif analyzer_priority_string == 'HIGH':
+      return Priority.HIGH
+    elif analyzer_priority_string == 'MEDIUM':
+      return Priority.MEDIUM
+    else:
+      return Priority.LOW
+
+  def brute_force_analysis(self, df: pd.DataFrame) -> Tuple[Priority, str, str]:
+    """Run brute force analysis"""
+    bfa = BruteForceAnalyzer()
+    bfa_result = bfa.run(df)
```

Comment:
We should probably catch `TurbiniaException` or whatever ends up getting raised by this call (or in the caller to the `brute_force_analysis` method).

Reply:
The ` BruteForceAnalyzer.run` will return `None` if unexpected conditions are met. I have added block to check if `bfa_result` is `None`. The `brute_force_analysis` returns Tuple regardless when unexpected conditions are met.

I don't anticipate this block with raise exception. However, happy to further discuss with you.

Reply:
This might have changed since you wrote this comment, but both the `run()` method and the `AnalyzerOutput.validate()` methods can throw `TurbiniaException`s so you might want to add a try/except for that here now.